java/apirag-core/src/main/java/com/datasqrl/ai/models/AbstractModelConfiguration.java

```diff
@@ -0,0 +1,65 @@
+package com.datasqrl.ai.models;
+
+import com.datasqrl.ai.util.ErrorHandling;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.configuration2.MapConfiguration;
+
+@AllArgsConstructor
+public abstract class AbstractModelConfiguration implements ModelConfiguration {
+
+  public static final String MODEL_NAME_KEY = "name";
+  public static final String MAX_INPUT_TOKENS_KEY = "max_input_tokens";
+  public static final String MAX_OUTPUT_TOKENS_KEY = "max_output_tokens";
+  public static final String TEMPERATURE_KEY = "temperature";
+  public static final double TEMPERATURE_DEFAULT = 0.7;
+  public static final String TOP_P_KEY = "top-p";
+  public static final double TOP_P_DEFAULT = 0.9;
+  public static final String TOKENIZER_KEY = "tokenizer";
```

Comment:
@mbroecheler Why should we make the tokenizer customizable? Most LLMs require a specific tokenizer and we choose the right one in our model type. I would not add this to the first version.